spec/controllers/queue_items_controller_spec.rb

```diff
@@ -2,138 +2,123 @@
-          delete :destroy, id: queue_item2.id
-          expect(queue_item1.reload.position).to eq(1)
-          expect(queue_item3.reload.position).to eq(2)
-        end
-
-        it "does not allow a user to delete a queue item that they do not own" do
-          bob = Fabricate(:user)
-          bobs_queue_item = Fabricate(:queue_item, video: superman, user: bob, position: 1)
-          delete :destroy, id: bobs_queue_item.id
-          expect(QueueItem.all).to eq([bobs_queue_item])
-        end
-      end
-
-      context "when user is unauthenticated" do
-        it "redirect to the sign in page" do
-          alice = Fabricate(:user)
-          video = Fabricate(:video)
-          queue_item = Fabricate(:queue_item, video: video, user: alice)
-          delete :destroy, id: queue_item.id
-          expect(response).to redirect_to(sign_in_path)
-        end
+    context "with authenticated users" do
+      before { set_current_user(alice) }
+
+      it "redirects to the my queue page" do
+        delete :destroy, id: alices_queue_item.id
+        expect(response).to redirect_to(my_queue_path)
+      end
+
+      it "deletes queue item" do
+        delete :destroy, id: alices_queue_item.id
+        expect(QueueItem.count).to eq(0)
+      end
+
+      it "normalizes the remaining queue items" do
+        video2 = Fabricate(:video)
+        queue_item1 = Fabricate(:queue_item, video: video, user: alice, position: 1)
+        queue_item2 = Fabricate(:queue_item, video: video2, user: alice, position: 2)
+        delete :destroy, id: queue_item1.id
+        expect(queue_item2.reload.position).to eq(1)
+      end
+
+      it "does not allow a user to delete a queue item that they do not own" do
+        bob = Fabricate(:user)
+        bobs_queue_item = Fabricate(:queue_item, video: video, user: bob, position: 1)
+        delete :destroy, id: bobs_queue_item.id
+        expect(QueueItem.all).to eq([bobs_queue_item])
       end
     end
   end
 
   describe "POST update_queue" do
-    context "with valid inputs" do
+    context "with valid inputs by an authenticated users" do
+      let(:video1) { Fabricate(:video) }
+      let(:video2) { Fabricate(:video) }
       let(:alice) { Fabricate(:user) }
-      let(:superman) { Fabricate(:video) }
-      let(:batman) { Fabricate(:video) }
-      let(:queue_item1) { Fabricate(:queue_item, video: superman, user: alice, position: 1) }
-      let(:queue_item2) { Fabricate(:queue_item, video: batman, user: alice, position: 2) }
 
       before do
-        session[:user_id] = alice.id
+        set_current_user(alice)
         post :update_queue, queue_items: [
           { id: queue_item1.id, position: 3 },
           { id: queue_item2.id, position: 2 }
         ]
       end
+      let(:queue_item1) { Fabricate(:queue_item, video: video1, user: alice, position: 1) }
+      let(:queue_item2) { Fabricate(:queue_item, video: video2, user: alice, position: 2) }
```

Comment:
I would add `let`'s before `before` block. Just to make it nicely organized and ordered.